cmd/juju/config/config.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package config
+
+import (
+	"strings"
+
+	"github.com/juju/cmd/v3"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/utils/v3/keyvalues"
+)
+
+// ConfigAction represents the action we want to perform here.
+// The action will be set in Init, and then accessed by the child command's
+// Run method to decide what to do.
+type ConfigAction string
+
+const (
+	Get     ConfigAction = "get"
+	Set     ConfigAction = "set"
+	SetFile ConfigAction = "set-file"
+	Reset   ConfigAction = "unset"
+)
+
+// ConfigCommand provides a common interface/functionality for configuration
+// commands (such as config and model-config). It defines SetFlags and Init
+// methods, while the individual command needs to define its own Run and Info
+// methods.
+type ConfigCommand struct {
+	// accept either modelcmd.ModelCommandBase or modelcmd.ControllerCommandBase
+	ModelCmd interface{ SetFlags(f *gnuflag.FlagSet) }
```

Comment:
I don't think we need this here, in go you can compose structs, so it make sense for the top level command struct to just call SetFlags on both this ConfigCommand and the ModelCommandBase/ControllerCommandBase